### PR TITLE
corruption alarm

### LINKED ISSUE #7125: alarm: CORRUPT
As part of integrity checking, etcd could raise a `CORRUPT` alarm if it detects potential cluster corruption.

### PR DESCRIPTION
**Pull request by heyitsanthony**
Adds grpc to peers to get the hashkv rpc.

Fixes #7125

### CODE REVIEW

**Comment by gyuho** [comment]
Would this cover https://github.com/coreos/etcd/issues/8313 as well?

**Comment by heyitsanthony** [comment]
@gyuho no, but would #8313 depend on it. This only does periodic checks by the leader.

**Comment by xiang90** [comment]
@heyitsanthony We probably should put this under an alpha feature flag initially?

File: etcdserver/apply.go [hunk 1]
```diff
@@ -577,9 +577,11 @@ func (a *applierV3backend) Alarm(ar *pb.AlarmRequest) (*pb.AlarmResponse, error)
 			break
 		}
 
+		plog.Warningf("alarm %v raised by peer %+v", m.Alarm, types.ID(m.MemberID))
```
Comment (gyuho) [inline etcdserver/apply.go]: Should just be `%s`? Becaue `types.ID` implements `String` method?

File: etcdserver/corrupt.go [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+		cli, cerr := clientv3.New(clientv3.Config{Endpoints: m.PeerURLs})
+		if cerr != nil {
+			continue
+		}
+
+		respsLen := len(resps)
+		for _, c := range cli.Endpoints() {
+			ctx, cancel := context.WithTimeout(context.Background(), s.Cfg.ReqTimeout())
+			resp, herr := cli.HashKV(ctx, c, rev)
+			fmt.Printf("got %+v %v\n", resp, err)
```
Comment (gyuho) [inline etcdserver/corrupt.go]: Log instead of print?

**Comment by heyitsanthony** [comment]
Modest thrashing on this one. Putting cmux on the peer port on the integration package caused trouble due to connections hanging in the matcher phase on shutdown. I vendored the mainline cmux since it has a read timeout which eventually cancels the read.

File: etcdmain/config.go [hunk 3]
```diff
@@ -216,6 +216,9 @@ func newConfig() *config {
 	// auth
 	fs.StringVar(&cfg.AuthToken, "auth-token", cfg.AuthToken, "Specify auth token specific options.")
 
+	// experimental
+	fs.DurationVar(&cfg.ExperimentalCorruptCheckTime, "experimental-corrupt-check-time", cfg.ExperimentalCorruptCheckTime, "Duration of time between cluster corruption check passes.")
```
Comment (gyuho) [inline etcdmain/config.go]: Also add this to https://github.com/coreos/etcd/blob/master/etcdmain/help.go?

**Pull request merged**